ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/MultiPartySubmissionIT.scala

```diff
@@ -0,0 +1,175 @@
+        )
+
+        // Exercise a choice to add (Charlie, David)
+        // Requires authorization from all four parties
+        _ <- ledger.exercise(
+          actAs = List(alice, bob, charlie, david),
+          readAs = List.empty,
+          exercise = contract
+            .exerciseAddSignatories(unusedActorArgument, Primitive.List(alice, bob, charlie, david))
+        )
+        active <- ledger.activeContracts(david)
+      } yield {
+        assert(active.length == 1)
+        assert(active.head.templateId.get == MultiPartyContract.id.unwrap)
+      }
+  })
+
+  test(
+    "MPSsubmitAndWaitMultiPartyChoiceReadDelegation",
+    "SubmitAndWait exercises a multi-party choice with read delegation",
+    allocate(Parties(4)),
+  )(implicit ec => {
+    case Participants(Participant(ledger, alice, bob, charlie, david)) =>
+      for {
+        // Create a contract for (Alice, Bob)
+        contract <- ledger.create(
+          actAs = List(alice, bob),
+          readAs = List.empty,
+          template = MultiPartyContract(Primitive.List(alice, bob), "")
+        )
+
+        // Exercise a choice to duplicate contract for (Charlie, David)
+        // Requires authorization from all the new parties,
+        // and read delegation from one of the old parties
+        _ <- ledger.exercise(
+          actAs = List(charlie, david),
+          readAs = List(alice),
+          exercise =
+            contract.exerciseDuplicateFor(unusedActorArgument, Primitive.List(charlie, david)))
+        active <- ledger.activeContracts(david)
+      } yield {
+        assert(active.length == 1)
+        assert(active.head.templateId.get == MultiPartyContract.id.unwrap)
+      }
+  })
+
+  test(
+    "MPSsubmitAndWaitMultiPartyChoiceMissingReadDelegation",
+    "SubmitAndWait should fail to exercise a multi-party choice with missing read delegation",
+    allocate(Parties(4)),
+  )(implicit ec => {
+    case Participants(Participant(ledger, alice, bob, charlie, david)) =>
+      for {
+        // Create a contract for (Alice, Bob)
+        contract <- ledger.create(
+          actAs = List(alice, bob),
+          readAs = List.empty,
+          template = MultiPartyContract(List(alice, bob), "")
+        )
+
+        // Exercise a choice to duplicate contract for (Charlie, David)
+        // Should fail, as the new parties do not see the original contract
+        failure <- ledger
+          .exercise(
+            actAs = List(charlie, david),
+            readAs = List.empty,
+            exercise =
+              contract.exerciseDuplicateFor(unusedActorArgument, Primitive.List(charlie, david)))
+          .failed
+      } yield {
+        assertGrpcError(
+          failure,
+          Status.Code.INVALID_ARGUMENT,
+          None,
+        )
+      }
+  })
+
+  // The "actor" argument in the generated methods to exercise choices is not used
+  private[this] val unusedActorArgument: Primitive.Party = Primitive.Party("")
```

Comment:
Juding from [here](https://github.com/digital-asset/daml/blob/master/language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/LFUtil.scala#L248) and [here](https://github.com/digital-asset/daml/blob/master/language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/TemplateCompanion.scala#L83), the actor argument in codegen exercise methods is not used for anything.

@S11001001 is that correct? If it's not used, do we keep that argument for backwards compatibility?